src/core/server/legacy_compat/logging/legacy_logging_server.ts

```diff
@@ -41,11 +41,20 @@ interface PluginRegisterParams {
  *  same as the rest of the records generated by the "legacy" Kibana. But to reduce
  *  overhead of having full blown Hapi server instance we create our own "light" version.
  */
-export class LegacyLoggingServer extends EventEmitter {
+export class LegacyLoggingServer {
   public connections = [];
+  // Emulates Hapi's usage of the podium event bus.
+  public events: Podium = new Podium();
+  private onPreStartCallbacks: Array<() => void> = [];
+  private onPostStopCallbacks: Array<() => void> = [];
 
   constructor(legacyLoggingConfig: Readonly<Record<string, any>>) {
-    super();
+    // Tell our podium instance about the events we'll be using.
+    this.events.registerEvent('log');
```

Comment:
Hapi replaced it's event bus with Podium (previously was an EventEmitter). This library requires that events that are emitted are first published. More deets here: https://github.com/hapijs/hapi/issues/3323

Reply:
Just played with this PR and dug into `good` source code and I think we can simplify this file a bit since `LegacyLoggingServer` shouldn't log any `ops` info so it can [just ignore `onPreStart` completely](https://github.com/hapijs/good/blob/master/lib/index.js#L17-L24). What about something like this?
```ts
export class LegacyLoggingServer {
  // Tell our podium instance about the events we'll be using. We don't actually
  // forward `request` and `response` events, but we have to register them with Podium
  // in order for the plugin to be able to setup.
  public events: Podium = new Podium(['log', 'request', 'response']);

  private onStopCallback?: () => void;

  constructor(legacyLoggingConfig: Readonly<Record<string, any>>) {
    setupLogging(
      this,
      Config.withDefaultSchema(transformDeprecations({ logging: legacyLoggingConfig }))
    );
  }
  .....
  public log({ level, context, message, error, timestamp, meta = {} }: LogRecord) {
    this.events.emit('log', { ... });
  }

  public stop() {
    if (this.onStopCallback !== undefined) {
      this.onStopCallback();
    }
  }

  public ext(eventName: ServerExtType, callback: () => void) {
    // We care only about stop callback here.
    if (eventName === 'onPostStop') {
      this.onStopCallback = callback;
    }
  }

  public expose() {....}
}

```

Reply:
Good point, I'll simplify this. Originally I was thinking about making this resilient to any changes we made to good, but I don't expect any and think it's pretty likely we'll drop it in the near future. FYI, we are using our fork of good at elastic/good.